source/game/system/map/MapdataCheckPoint.hh

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "game/system/map/MapdataAccessorBase.hh"
+
+#include <egg/math/Vector.hh>
+
+namespace System {
+
+class MapdataCheckPoint;
+class MapdataCheckPointAccessor;
+
+struct LinkedCheckpoint {
+    MapdataCheckPoint *checkpoint;
+    EGG::Vector2f p0diff;
+    EGG::Vector2f p1diff;
+    f32 distance;
+};
+
+class MapdataCheckPoint {
+public:
+    static constexpr s8 NORMAL_CHECKPOINT = -1; ///< only used for picking respawn position
+    static constexpr s8 FINISH_LINE = 0;        ///< triggers a lap change; also the starting line
+
+    struct SData {
+        EGG::Vector2f left;
+        EGG::Vector2f right;
+        u8 jugemIndex;
+        s8 lapCheck;
+        u8 prevPt;
+        u8 nextPt;
+    };
+
+    enum class SectorOccupancy {
+        InsideSector,  ///< if player is inside the checkpoint quad
+        OutsideSector, ///< if player is not between the sides of the quad (may still be between
+                       ///< this checkpoint and next); player is likely in a different checkpoint
+                       ///< group
+        OutsideSector_BetweenSides, ///< if player is between the sides of the quad, but NOT between
+                                    ///< this checkpoint and next; player is likely in the same
+                                    ///< checkpoint group
+    };
+
+    MapdataCheckPoint(const SData *data);
+    void read(EGG::Stream &stream);
+
+    bool isPlayerFlagged(s32 playerIdx) const;
+    void setPlayerFlags(s32 playerIdx);
+    void resetFlags();
+
+    /// @beginGetters
+    [[nodiscard]] const EGG::Vector2f &left() const;
+    [[nodiscard]] const EGG::Vector2f &right() const;
+    [[nodiscard]] u8 jugemIndex() const;
+    [[nodiscard]] s8 type() const;
+    [[nodiscard]] bool isNormalCheckpoint() const;
+    [[nodiscard]] bool isFinishLine() const;
+    [[nodiscard]] u16 nextCount() const;
+    [[nodiscard]] u16 prevCount() const;
+    [[nodiscard]] u16 id() const;
+    [[nodiscard]] MapdataCheckPoint *prevPoint(s32 i) const;
+    [[nodiscard]] MapdataCheckPoint *nextPoint(s32 i) const;
+    [[nodiscard]] const LinkedCheckpoint &nextLinked(s32 i) const;
```

Comment:
```suggestion
    [[nodiscard]] bool isNormalCheckpoint() const;
    [[nodiscard]] bool isFinishLine() const;
```
You don't need most of these since the only thing that accesses them right now is `thisptr`